front/src/applications/stdcm/components/SimulationReportSheet.tsx

```diff
@@ -32,6 +34,33 @@ const SimulationReportSheet = ({
     path_number2: 'n°YYYYYY',
   };
 
+  const [mapBlob, setMapBlob] = useState<string | null>(null);
+
+  // Convert image to JPEG
+  useEffect(() => {
+    let objectUrl: string | null = null;
+
+    if (mapCanvas) {
+      base64ToJpeg(mapCanvas, 0.8).then((blob) => {
+        if (blob) {
+          // Revoke the previous object URL if it exists
+          if (objectUrl) {
+            URL.revokeObjectURL(objectUrl);
+          }
+          objectUrl = URL.createObjectURL(blob);
+          setMapBlob(objectUrl);
+        }
+      });
+    }
+
+    // Cleanup the object URL when the component is unmounted or before a new one is created
+    return () => {
+      if (objectUrl) {
```

Comment:
Instead of a separate `objectUrl` variable, we could register a callback which runs on unmount in a separate `useEffect()` and uses the state variable. Could be simpler/more readable.

Reply:
Fixed